lib/elixir/src/elixir_expand.erl

```diff
@@ -1008,6 +1023,20 @@ format_error({op_ambiguity, Name, Arg}) ->
     "\"~ts ~ts\" looks like a function call but there is a variable named \"~ts\", "
     "please use explicit parentheses or even spaces",
   io_lib:format(Message, [Name, 'Elixir.Macro':to_string(Arg), Name]);
+format_error({invalid_clauses, Name}) ->
+  SupportMessage =
+    case Name of
+      'case' ->
+        "If you are trying to use the \"case\" structure, it should be similar to:\n"
+        "    case value do"
+        "      pattern -> :ok"
+        "    end";
+
+      _ ->
+        "Clauses are only valid for case, cond, receive, catch, rescue and fn."
```

Comment:
This is not entirely true. It depends - e.g. in `with`, it's also valid with `else`.

Reply:
I've added `else` to the list, I can't figure out a better message without getting too complicated.